### ISSUE TITLE
(Settings/Displays): Using Orientation Flipped mirrors the content on the Y axis

### ISSUE THREAD

**Issue by ahoneybun** [pop-os/cosmic-settings#166]
Using either Orientation Flipped setting flips the content incorrectly.

![PXL_20240226_214157318](https://github.com/pop-os/cosmic-settings/assets/4884946/e5aeee1b-31d2-46ba-94f0-c1bf4fbfe6c5)

**Comment by XV-02** [comment]
![screenshot-2024-02-26-21-46-11](https://github.com/pop-os/cosmic-settings/assets/98765732/ec2335aa-1609-4477-961e-be992d5f92f7)

I can second seeing this on recent releases. Normally, the "flipped" orientation is effectively rotated through 180 (which I think is the same as a horizontal mirror followed by a vertical mirror, but my spacial awareness fails me just now).

However, on Cosmic DE at current, a "flipped" orientation is only resulting in horizontal mirroring. 

Interestingly, when I went to take the screenshot above of a flipped portrait view, the content I was screenshotting un-flipped, but the screenshot tool itself was rendered as mirrored atop the now unmirrored content.

**Comment by XV-02** [comment]
Also, this is weird: Every time I hit the `print-screen` key, with `cosmic-screenshot` already launched, the image is flipped over the Y-axis. 

So, first time I hit `print-screen` key, the screenshot tool launches. Without dismissing the tool, I hit `print-screen` again, and the content flips - including flipping the image that is recorded by `cosmic-screenshot`. If I hit `print-screen` a third time without taking a screenshot, then the content flips back to being mirrored. This is only an issue when I use one of the "flipped" display settings (landscape or portrait).

**Comment by mmstick** [comment]
The option is named after its Wayland protocol variant, which it sets. So it was working as intended by the standard. There are additional transform modes that I could expose in the settings UI, such as `rotate180`.

**Comment by XV-02** [comment]
That would probably be helpful.

**Comment by leviport** [comment]
Should mirrored modes even be exposed in display settings? Seems a bit more advanced than what most users would need...

**Comment by mmstick** [comment]
I think it would make sense to have just rotate options in the dropdown, and then a flipped toggle next to it.

**Comment by XV-02** [comment]
So, the rotated displays are important as I imagine 50% of the time, a portrait orientation needs rotation, and I have definitely seen set ups with larger monitors where people use a rotated landscape output for a second vertically stacked monitor - often when the monitor has a thin bezel, so they have the thinnest possible divide between their displays. 

The combination of rotated and flipped features gives us, effectively, every realistic orientation of a screen. I don't often see mirrored displays used, but I can imagine situations where the ability to "back-project" would be valuable - like my optometrist who uses a monitor and a mirror to get the correct simulated distance for an eye chart in their tiny patient room.

**Comment by leviport** [comment]
I really think we should consult @maria-komarova on this one. While I'm biased towards exposing more functionality, I think the use cases for flipped monitors would be exceptionally rare. Users who need such an exotic setup probably won't have a lot of qualms with being told to install wdisplays to achieve it.

**Comment by maria-komarova** [comment]
I don't think we need flipped in the UI. Doesn't seem to me like the frequently used functionality. Or we need some data that says it is used frequently. 
What is the rotated feature? Or is it the same?

**Comment by ahoneybun** [comment]
@maria-komarova the rotated feature would be similar to this:

![Screenshot from 2024-02-28 10-50-41](https://github.com/pop-os/cosmic-settings/assets/4884946/4d5b56b1-d97c-4d7f-bd87-397bebb865f1)

**Comment by leviport** [comment]
But "flipped" in that Gnome example means rotated 180°. The "flipped" that we're talking about is akin to mirroring the output, and is independent of rotation.

**Comment by mmstick** [comment]
There are 8 options: `normal`, `rotate90`, `rotate180`, `rotate270`, `flipped`, `flipped90`, `flipped180`, `flipped270`. Flipped mirrors the image. There may be use cases for flipped when outputting to a projector that is mounted upside down on the ceiling.

**Comment by maria-komarova** [comment]
If we do rotation we should probably use angles, like 90, 180, 270.

**Comment by maria-komarova** [comment]
Let's do rotation with rotate 90, rotate 180, rotate 270. And let's skip flipped options for now.

**Comment by mmstick** [comment]
@maria-komarova Should we change the `Landscape` and `Portrait` descriptions to `Normal`, `Rotate 90`, `Rotate 180`, and `Rotate 270`?

**Comment by maria-komarova** [comment]
Standard, rotate 90, rotate 180, rotate 270.
